Avoid force-unwrapping an optional

geoffreygaren opened the pull request:
#### e642a73660e17631b30ae59ace74836bd43aabd0
<pre>
Avoid force-unwrapping an optional
<a href="https://bugs.webkit.org/show_bug.cgi?id=300931">https://bugs.webkit.org/show_bug.cgi?id=300931</a>
<a href="https://rdar.apple.com/162807734">rdar://162807734</a>

Reviewed by Mike Wyrzykowski.

Swift 5.7 if-let syntax enables us to rule out null deref syntactically.

<a href="https://github.com/swiftlang/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md">https://github.com/swiftlang/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md</a>

* Source/WebGPU/WebGPU/CommandEncoder.swift:
(WebGPU.beginRenderPass(_:)):
(WebGPU.beginComputePass(_:)):

Canonical link: <a href="https://commits.webkit.org/301683@main">https://commits.webkit.org/301683@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b90beee6064df4b882ec7da72b956c0f51d84461

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126682 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46325 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37288 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133644 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78332 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f7b3c2ca-2534-492d-aea8-b85f7b77eb78) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46957 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54863 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96398 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64462 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/26ce671e-20e7-4439-ac1a-8ca426018173) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129630 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37570 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113307 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76923 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/48c36687-ed9d-49e8-b537-cf9988c0c836) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36456 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77041 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107385 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31790 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136219 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53367 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41061 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104915 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53857 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109667 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104616 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50107 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28448 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50806 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19827 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53291 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59097 "Failed to build and analyze WebKit") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52560 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55896 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54300 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebGPU/WebGPU/CommandEncoder.swift:
```diff
@@ -833,7 +833,7 @@ extension WebGPU.CommandEncoder {
         }
 
-        if m_commandBuffer != nil && m_commandBuffer!.status.rawValue >= MTLCommandBufferStatus.enqueued.rawValue {
+        if let m_commandBuffer, m_commandBuffer.status.rawValue >= MTLCommandBufferStatus.enqueued.rawValue {
```
if m_commandBuffer is nil, it will not go in this branch. Is that expected?

mwyrzykowski replied:
Yes, because one can not commit a nil command buffer to the command queue

rr-codes replied:
Later in the function, it calls `Device::trackTimestampsBuffer`, which seems to assume a non-nil value, but from this logic it looks like it can be passed in a nil value, resulting in calling objectForKey:nil which would cause a crash? Or am I misunderstanding?

rr-codes replied:
oh nvm, calling objectForKey on a nil value doesn’t crash

mwyrzykowski replied:
Yes, just to add `trackTimestampsBuffer` is a debugging functionality and potentially buggy in corner cases. Though of course we should fix the bugs if found.

geoffreygaren committed:
Avoid force-unwrapping an optional
https://bugs.webkit.org/show_bug.cgi?id=300931
rdar://162807734

Reviewed by Mike Wyrzykowski.

Swift 5.7 if-let syntax enables us to rule out null deref syntactically.

https://github.com/swiftlang/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md

* Source/WebGPU/WebGPU/CommandEncoder.swift:
(WebGPU.beginRenderPass(_:)):
(WebGPU.beginComputePass(_:)):

Canonical link: https://commits.webkit.org/301683@main
